src/modules/adaptors/Frax/DebtFTokenAdaptor.sol

```diff
@@ -0,0 +1,310 @@
+        // Check if borrower is insolvent after this borrow tx, revert if they are
+        if (!_isSolvent(address(this), _exchangeRate)) {
+            revert DebtFTokenAdaptor__LTVTooLow(address(fToken));
+        }
+    }
+
+    // `repayDebt`
+
+    /**
+     * @notice Allows strategists to repay loan debt on Fraxlend Pair.
+     * @dev Uses `_maxAvailable` helper function, see BaseAdaptor.sol
+     * @param tokenToRepay the underlying ERC20 token you want to repay, NOT the debtToken. TODO: not sure if we want this to be pulling from the fraxlendPair the assetToken, or to just have this be FRAX since it always is FRAX and doesn't sound like it will change. NOTE that the assetContract is actually internal var within fraxlend pair.
+     * @param amountToRepay the amount of `tokenToRepay` to repay with.
+     * NOTE: TODO: CRISPY QUESTION - this is the simpler version but I think we ought to go with something closer to the alternative (see comments in next function).
+     */
+    function repayFraxlendDebt(
+        IFToken fToken,
+        ERC20 tokenToRepay,
+        uint256 sharesToRepay
+    ) public {
+        uint256 fraxlendRepayAmount = _toBorrowAmount(fToken, sharesToRepay, false, ACCOUNT_FOR_INTEREST);
+        tokenToRepay.safeApprove(address(fToken), fraxlendRepayAmount);
+        fToken.repayAsset(sharesToRepay, msg.sender);
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(tokenToRepay, address(fToken));
+    }
+
+    /**
+     * @notice Allows strategists to repay loan debt on Fraxlend Pair.
+     * @dev Uses `_maxAvailable` helper function, see BaseAdaptor.sol
+     * @param tokenToRepay the underlying ERC20 token you want to repay, NOT the debtToken. TODO: not sure if we want this to be pulling from the fraxlendPair the assetToken, or to just have this be FRAX since it always is FRAX and doesn't sound like it will change.
+     * @param amountToRepay the amount of `tokenToRepay` to repay with.
+     * NOTE: this version of repayFraxlendDebt is meant to offer an alternative to doing it similar to the AaveDebtTokenAdaptor (see notes below)
+     * TODO: CRISPY --> from `FraxlendCore.sol` functions `repayAsset()` and `_repayAsset()` I don't see specific responses from the contracts if the amount of shares repaid >> amount of borrowShares even owed. Unless `-=` takes care of that.
+     * Assuming the above is true, then we'll need to check how much borrowShares we need to pay. Then we can get the total amount of Frax we need to repay. So we could specify the amount of FRAX we're willing to repay (could be max), and then we calculate the amount of FRAX owing from fraxlend, then we check the two values to make sure it is less than the amount we're willing to repay. From there we can simply repay the shares. We then check that we have the expected change in borrowShares as per FraxLend accounting.
+     */
+    function repayFraxlendDebt2(
```

Comment:
I think this one is better. I want to add some new logic to all of our lending adaptors, repay max debt, which either caps it to the debt owed, or it is capped by the ERC20s the cellar has.

Like if 10k is owed, but the cellar only has 8k, the logic should have it repay 8k of debt

Reply:
kk, sounds good. I'll have this one be the live version of the function.